quickannotator/api/v1/annotation/utils.py

```diff
@@ -0,0 +1,117 @@
+#%%
+import threading
+import numpy as np
+import ray
+import time
+import os
+from datetime import datetime
+
+from quickannotator.db import get_session
+from quickannotator.db.utils import build_export_filepath
+from quickannotator.db.crud.annotation import AnnotationStore
+from quickannotator.dsa_sdk import DSAClient
+import geojson
+from quickannotator import constants
+from quickannotator.db.crud.image import get_image_by_id
+from itertools import product
+
+
+class ProgressTracker:
+    def __init__(self, total: int):
+        self.total = total
+        self.progress = 0
+        self.lock = threading.Lock()  # Add a threading lock for thread safety
+
+    def increment(self):
+        with self.lock:  # Ensure thread safety by locking during increment
+            self.progress += 1
+
```

Comment:
technically speaking with this design its possible to get a progress > total, is that something to be concerned about? my intuition is no and its the "clients" job to keep track of their stuff and manage the state "correctly"?

Reply:
you see this one?